app/code/community/Bolt/Boltpay/Model/Cron.php

```diff
@@ -66,17 +66,17 @@ public function cleanupQuotes() {
      */
     public function cleanupOrders() {
         try {
-            $expiration_time = Mage::getSingleton('core/date')
-                ->gmtDate('Y-m-d H:i:s', time()-(60*PRE_AUTH_STATE_TIME_LIMIT_MINUTES));  // Magento uses GMT to save in DB
+            $expiration_time = gmdate('Y-m-d H:i:s', time()-(60*self::PRE_AUTH_STATE_TIME_LIMIT_MINUTES));  // Magento uses GMT to save in DB
 
             /* @var Mage_Sales_Model_Resource_Order_Collection $orderCollection */
             $orderCollection = Mage::getModel('sales/order')->getCollection();
 
+             $orderCollection
+                ->addFieldToFilter('created_at', array( 'gteq' => $expiration_time))
+                ->setOrder('created_at', 'ASC');
+
             /** @var Mage_Sales_Model_Order $deletePendingPaymentOrdersBeforeThis */
-            $deletePendingPaymentOrdersBeforeThis = $orderCollection
-                ->addFieldToFilter('created_at', array( 'gte' => $expiration_time))
-                ->setOrder('created_at', 'ASC')
-                ->getFirstItem();
+            $deletePendingPaymentOrdersBeforeThis = $orderCollection->getFirstItem();
```

Comment:
Is there a reason why you split this instead of adding a single "q"?

Reply:
@adentong 
There a few factors into this.

Firstly, in M1, by default, Bolt orders that have not been paid for are created without a timestamp.  This serves to hide the order from ERPs that import orders base on the `created_at` timestamp.  Because of this, we use the timestamps of converted orders to find the lowest id from which it is safe to delete Bolt pending payment orders.

Second, we are interested in order model objects.  When we delete, we want to use the ORM so that we trigger all the  observers of the deletion event so that they execute their code and the removal of the orders is cleaned up the right way for all 3rd party plugins.

This brings me to the third factor.  If I were simply writing a plain SQL query to get the ids of the orders, then it would be a "not-even-thinking-twice-about-it" writing a single SQL statement.  However, since I am using their ORM and query builder, making the joins using it seems much more complex and harder to understand for future developers than to breaking things down into two very simple and understandable collection object queries.  Moreover, I am trusting that their implementation of this is more efficient than me first querying for the order IDs and then loading the order objects in a loop one-by-one in order to delete them.

So, for the the end goal, I believe like this is simpler to understand and easier to implement and easier to maintain.